webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -182,23 +205,171 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
   const logFilters = (
     <LogViewerSelect
       options={filterOptions}
-      showSearch={true}
+      searchOn={searchOn}
+      showSearch={false}
       values={filterValues}
       onChange={handleFilterChange}
+      onClickSearch={() => setSearchOn((prev) => !prev)}
       onReset={handleFilterReset}
     />
   );
 
+  const throttledChangeSearch = useMemo(
+    () =>
+      throttle(
+        500,
+        (s: string) => {
+          updateSettings({ ...settings, searchText: s });
+        },
+        { noLeading: true },
+      ),
+    [updateSettings, settings],
+  );
+
+  useEffect(() => {
+    return () => {
+      throttledChangeSearch.cancel();
```

Comment:
it might make sense to commit the last known value to the settings before unmount -- we've run into cases where the settings value is not what we expect given a throttled update in e2e tests. Maybe something like this should work?
```suggestion
      throttledChangeSearch.cancel();
      // kinda gross but we want this to run only on unmount
      setSearchInput((s) => updateSettings({ ...settings, searchText: s }) || s);
```